test/Concurrency/DeferredSendableChecking/region_based_sendability.swift

```diff
@@ -670,3 +670,38 @@ func many_consume_many_require_vararg(a : A) async {
     foo_noniso_vararg(ns6, ns1, ns7); //expected-note {{access here could race}}
     foo_noniso_vararg(ns7, ns6, ns2); //expected-note {{access here could race}}
 }
+
+enum E {
+    case E1(NonSendable)
+    case E2(NonSendable)
+    case E3(NonSendable)
+}
+
+func enum_test(a : A) async {
+    let e1 = E.E1(NonSendable())
+    let e2 = E.E2(NonSendable())
+    let e3 = E.E3(NonSendable())
+    let e4 = E.E1(NonSendable())
+
+    switch (e1) {
+    case let .E1(ns1):
+        switch (e2) {
+        case let .E2(ns2):
+            ns1.x = ns2.x
+        default: ()
+        }
+    case .E2:
+        switch (e3) {
+        case let .E3(ns3):
+            await a.foo(ns3.x); //expected-warning{{passing argument of non-sendable type 'Any' from nonisolated context to actor-isolated context at this call site could yield a race with accesses later in this function (1 access site displayed)}}
+        default: ()
+        }
+    case .E3:
+        foo_noniso(e4);
+    }
+
+    await a.foo(e1); //expected-warning{{passing argument of non-sendable type 'E' from nonisolated context to actor-isolated context at this call site could yield a race with accesses later in this function (1 access site displayed)}}
+    foo_noniso(e2); //expected-note{{access here could race}}
+    foo_noniso(e3); //expected-note{{access here could race}}
```

Comment:
Just to check my understanding here, the warning on line 696 corresponds to the note on line 705, right? That is, you're able to detect that when `ns3.x` is sent, that's also considered a send `e3` since `ns3` came from its payload.

Also, neat how the warning on line 703 for a send of `e1` is was able to tell how it's related to `e2` because of line 690. 

Worth considering adding a comment to this test case explaining what's happening here a bit, since it took a minute for me at least.